# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSink.cpp
# Commit: 3fbc1456c6d2
# Full Hash: 3fbc1456c6d26d91729498ea6ad19dc24e479dfa
# Author: Matthew Gregan <kinetik@flim.org>
# Date: 2019-10-02 03:38:52
# Regressor Bug: 1544997
# File Overlap Count: 1
# Description:
#   Bug 1544997 - Propogate cubeb stream error to AudioSink to handle loss of audio device.  r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D47745
# ==============================================================================

diff -r f53153babb6f -r 3fbc1456c6d2 dom/media/mediasink/AudioSink.cpp
--- a/dom/media/mediasink/AudioSink.cpp	Tue Oct 01 20:01:54 2019 +0000
+++ b/dom/media/mediasink/AudioSink.cpp	Tue Oct 01 12:14:44 2019 +0000
@@ -301,6 +301,12 @@
   mEndedPromise.ResolveIfExists(true, __func__);
 }
 
+void AudioSink::Errored() {
+  SINK_LOG("Errored");
+  mPlaybackComplete = true;
+  mEndedPromise.Reject(NS_ERROR_FAILURE, __func__);
+}
+
 void AudioSink::CheckIsAudible(const AudioData* aData) {
   MOZ_ASSERT(aData);
 